### PR TITLE
Mock remote request in restapi to prevent timeouts when hitting api.wordpress.org endpoint.

### PR DESCRIPTION
**Pull request by hellofromtonya**
Recent timeout failures ([see this job](https://github.com/WordPress/wordpress-develop/actions/runs/1447262178)) are due to remote requests not being mocked. This patch mocks the specific failing tests, though each test will need to be investigated separately for mocking strategy.

Trac ticket: https://core.trac.wordpress.org/ticket/54420

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@TimothyBJacobs This PR mocks the 2 failing tests we saw https://wordpress.slack.com/archives/C02RQBWTW/p1636603357442100

```
❗ A run of the PHPUnit Tests workflow has failed for the trunk branch.

REST API: Add /wp/v2/block-navigation-areas endpoint

More details: https://github.com/WordPress/wordpress-develop/actions/runs/1447262178.
```

The ticket I opened takes the slow road to fix each of these as found.

Can you do a confidence check of the mocks?

File: tests/phpunit/tests/rest-api/rest-plugins-controller.php [hunk 1]
```diff
@@ -542,10 +542,29 @@ public function test_create_item_wdotorg_unreachable() {
...
+						'message' => 'Not Found',
+					),
+					'body'     => '{"error":"Plugin not found."}',
+					'cookies'  => array(),
+					'filename' => null,
+				);
+			}
+		);
 
 		// This will hit the live API.
```
Comment (TimothyBJacobs) [inline tests/phpunit/tests/rest-api/rest-plugins-controller.php:569]: Let's remove this comment now.

File: tests/phpunit/tests/rest-api/rest-plugins-controller.php [hunk 2]
```diff
@@ -542,10 +542,29 @@ public function test_create_item_wdotorg_unreachable() {
 	}
 
 	/**
-	 * @ticket 50321
+	 * @ticket 50321x
```
Comment (TimothyBJacobs) [inline tests/phpunit/tests/rest-api/rest-plugins-controller.php:545]:
```suggestion
	 * @ticket 50321
```

**Review by TimothyBJacobs — APPROVED** [review]
lgtm!

File: tests/phpunit/tests/rest-api/rest-plugins-controller.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/rest-api/rest-plugins-controller.php]: Removed in https://github.com/WordPress/wordpress-develop/pull/1868/commits/ad0d8b39c87ae556253cfccec2d8a31a9c377675.

**Comment by hellofromtonya** [comment]
Committed via changesets https://core.trac.wordpress.org/changeset/52137 and https://core.trac.wordpress.org/changeset/52138.

**Pull request closed without merging**